graphql_compiler/query_pagination/__init__.py

```diff
@@ -81,17 +82,32 @@ def paginate_query_ast(
     remainder_queries: Tuple[ASTWithParameters, ...] = tuple()
     advisories: Tuple[PaginationAdvisory, ...] = tuple()
 
+    # Split the query if we should and we can
     result_size = query_analysis.cardinality_estimate
     num_pages = _estimate_number_of_pages(
         query_analysis.query_string_with_parameters, result_size, page_size
     )
     if num_pages > 1:
         pagination_plan, advisories = get_pagination_plan(query_analysis, num_pages)
-        if pagination_plan.vertex_partitions:
-            page_query, remainder_query = split_into_page_query_and_remainder_query(
-                query_analysis, pagination_plan
+        if len(pagination_plan.vertex_partitions) == 0:
+            pass
+        elif len(pagination_plan.vertex_partitions) == 1:
+            parameter_generator = generate_parameters_for_vertex_partition(
+                query_analysis.schema_info,
+                query_analysis.ast_with_parameters,
+                pagination_plan.vertex_partitions[0],
+            )
+            first_param = next(parameter_generator, None)
+            if first_param:
```

Comment:
Do we want some sort of advisory if we are unable to paginate as requested? Or is that not worth it / not the right idea for some reason?

Reply:
We use advisories to see what is worth fixing. We know this is worth fixing, so I'd rather fix it than spend more time on getting more signal.